README.md

```diff
@@ -66,19 +66,22 @@ To learn more about the different rule types, their guarantees and how they are
 #### Carrying out software supply chain steps
 
 ##### in-toto-run
-`in-toto-run` generates link metadata for the given command-line option and runs it as its own command. See the [this simple usage example from the demo application for more details](https://github.com/in-toto/demo#tampering-with-the-software-supply-chain).
-
-```shell
-in-toto-run  --step-name <unique step name>
-            {--key <functionary signing key path>,  --gpg [<functionary gpg signing key id>]}
-            [--gpg-home <path to gpg keyring>]
-            [--materials <filepath>[ <filepath> ...]]
-            [--products <filepath>[ <filepath> ...]]
-            [--record-streams]
-            [--no-command]
-            [--verbose] -- <cmd> [args]
-```
-
+`in-toto-run` is used to execute a step in the software supply chain. This can be
+anything relevant to the project such as tagging a release with `git`, running
+a test, or building a binary. The relevant step name and command are
+passed as arguments, along with materials, which are files required for that
+step's command to execute, and products which are files expected as a result
+of the execution of that command. These, and other relevant details
+pertaining to the step are stored in a link file, which is signed using the
+functionary's key. If materials are not passed to the command, the link file
+generated just doesn't record them. Similarly, if the execution of a command
+via `in-toto-run` doesn't result in any products, they're not recorded in the
+link file.
```

Comment:
IMHO the parts about materials and products are still ambiguous. The command reads/modifies/creates files no matter what the functionary passed as `--materials` or `--products`. And the reverse is true, in-toto records the hashes of passed `--materials` and `--products` no matter which files the command touches. So what is passed here is not immediately relevant for the command to execute, but it allows the functionary to explicitly provide information about the command execution, which in turn allows us to verify eventually that the steps of the supply chain are properly linked together. Does this make sense? Could you try once more to distill this information? :) Thanks!

Reply:
I'll give it a shot.